scri/asymptotic_bondi_data/transformations.py

```diff
@@ -145,9 +145,8 @@ def Bprm_j_k(thetaprm, phiprm):
     for j in range(n_theta):
         for k in range(n_phi):
             thetaprm_j, phiprm_k = thetaprm_phiprm[j, k]
-            R_j_k[j, k] = (
-                Bprm_j_k(thetaprm_j, phiprm_k) * frame_rotation * quaternion.from_spherical_coords(thetaprm_j, phiprm_k)
-            )
+            rotated_coord_quaternion = frame_rotation * quaternion.from_spherical_coords(thetaprm_j, phiprm_k)
+            R_j_k[j, k] = Bprm_j_k(*quaternion.as_spherical_coords(rotated_coord_quaternion)) * rotated_coord_quaternion
```

Comment:
I agree that this is the right thing to do.  Thanks for finding this!

Reply:
No problem! Glad we agree.